File: data/code_pairs/cpp/1549.cpp
#include <vector>
#include <iostream>
#include <cmath>
#include <cassert>

void COMPRO(std::vector<double>& PROFILE, const std::vector<double>& EMINT, int NFOBR, double W, int JFIRST, int JLAST) {
    for (int K = 0; K < NFOBR; ++K) {
        if (JFIRST == JLAST) {
            PROFILE[K] = EMINT[K];
        } else {
            PROFILE[K] = PROFILE[K] + W * EMINT[K];
        }
    }
}

File: data/code_pairs/fortran/1549.f90
module MOD_COMPRO
    contains
    subroutine COMPRO (PROFILE,EMINT,NFOBR,W,JFIRST,JLAST)
    implicit none
    real*8, intent(inout) :: PROFILE(:)
    real*8, intent(in) :: EMINT(:)
    real*8, intent(in) :: W
    integer, intent(in) :: NFOBR, JFIRST, JLAST
    integer :: K

    do K = 1, NFOBR
        if (JFIRST .EQ. JLAST) then
            PROFILE(K)=EMINT(K)
        else
            PROFILE(K)=PROFILE(K)+W*EMINT(K)
        endif
    enddo

    end subroutine COMPRO
end module MOD_COMPRO